### ISSUE TITLE
setValue has no way to bypass dirtying the form

### ISSUE THREAD

**Issue by vektah** [react-hook-form/react-hook-form#334]
**Is your feature request related to a problem? Please describe.**
When using custom components like ReactSelect it is required to be able to push the initial value of the control into form state without dirtying the form.

**Describe the solution you'd like**
Something along the lines of https://github.com/react-hook-form/react-hook-form/commit/9c50dd9567d3866bfb483bfde507230a415be792 would be fine. 

The third arg to setValue probably should be an option bag to avoid breaking the interface every time a new param is added.

**Describe alternatives you've considered**
We could also set the initial value during register.

**Additional context**

see also https://github.com/react-hook-form/react-hook-form/issues/72 which added setDirty to setValue.

**Comment by bluebill1049** [comment]
> The third arg to setValue probably should be an option bag to avoid breaking the interface every time a new param is added.

Third argument is optional.

also, I am not clear on your questions

**Comment by bluebill1049** [comment]
you should set the initial value at react-select, it's a controlled component. 

curious to know what's the use case when setValue shouldn't dirty the form?

**Comment by bluebill1049** [comment]
I don't see much need of changing `setValue` API for now, feel free to re-open, if you can provide a valid use case :) closing for now

**Comment by svenlombaert** [comment]
@bluebill1049 I came across this exact use case, and I can provide an example!

I have a form with a date and price field.
When a user changes the date, we fetch the latest price, to prefill the price field, so I call setValue to set that price. We want to keep doing this, until the user **actually** touched the price field and filled it in himself. Would be nice to have a way to bypass dirtying when setting dynamic default values like this.

**Comment by bluebill1049** [comment]
use reset and omitStateReset update https://react-hook-form.com/api#reset

**Comment by svenlombaert** [comment]
That seems to work, but the watchedFiels then all return undefined?

**Comment by bluebill1049** [comment]
@svenlombaert can you provide a CSB?

**Comment by svenlombaert** [comment]
@bluebill1049 Sure! I found out it only happens when I provide the omitState object, I do that, because I want to keep the dirty and dirtyFields state.

https://codesandbox.io/s/react-hook-form-custom-reset-ucff4?file=/src/App.js
(My bug is reacty-native, so that's why I made the inputs controlled)

To reproduce:
1.   fill in name and last name
2.  press 'reset with omitState'
3. values are undefined

If you do the above with the reset without omitState, it seems to work.
